packages/mux-video/src/index.ts

```diff
@@ -501,6 +554,21 @@ class MuxVideoElement extends CustomVideoElement<HTMLVideoElement> implements Pa
             .catch(() => console.error(`Unable to load or parse metadata JSON from metadata-url ${newValue}!`));
         }
         break;
+      case Attributes.STREAM_TYPE:
+        // If the newValue is unset
+        if (newValue == null || newValue !== oldValue) {
+          this.dispatchEvent(
+            new CustomEvent('streamtypechange', { composed: true, bubbles: true, detail: this.streamType })
```

Comment:
native video events don't have detail / data attached. not sure if it's the best idea to change expectations for our custom events (and I see I did that too for the error event 😄🔫) 

```suggestion
            new CustomEvent('streamtypechange', { composed: true, bubbles: true })
```

Reply:
Accepting change, but I believe this one should be a followup conversation. It is valid to have native events with specific instance-based properties (see many examples here https://developer.mozilla.org/en-US/docs/Web/API/Event#interfaces_based_on_event), but, if we want to implement those (for media-ui-extension proposals like this, for example), we should probably avoid `CustomEvent` and instead implement our own event class